kornia/utils/helpers.py

```diff
@@ -30,3 +30,67 @@ def _extract_device_dtype(tensor_list: List[Optional[Any]]) -> Tuple[torch.devic
     if dtype is None:
         dtype = torch.get_default_dtype()
     return (device, dtype)
+
+
+def _torch_inverse_cast(input: torch.Tensor) -> torch.Tensor:
+    """Helper function to make torch.inverse work with other than fp32/64.
+
+    The function torch.inverse is only implemented for fp32/64 which makes
+    impossible to be used by fp16 or others. What this function does, is cast
+    input data type to fp32, apply torch.inverse, and cast back to the input dtype.
+    """
+    assert isinstance(input, torch.Tensor), f"Input must be torch.Tensor. Got: {type(input)}."
+    dtype: torch.dtype = input.dtype
+    if dtype not in (torch.float32, torch.float64,):
+        dtype = torch.float32
+    return torch.inverse(input.to(dtype)).to(input.dtype)
+
+
+def _torch_histc_cast(input: torch.Tensor, bins: int, min: int, max: int) -> torch.Tensor:
+    """Helper function to make torch.histc work with other than fp32/64.
+
+    The function torch.histc is only implemented for fp32/64 which makes
+    impossible to be used by fp16 or others. What this function does, is cast
+    input data type to fp32, apply torch.inverse, and cast back to the input dtype.
+    """
+    assert isinstance(input, torch.Tensor), f"Input must be torch.Tensor. Got: {type(input)}."
+    dtype: torch.dtype = input.dtype
+    if dtype not in (torch.float32, torch.float64,):
+        dtype = torch.float32
+    return torch.histc(input.to(dtype), bins, min, max).to(input.dtype)
+
+
+def _torch_svd_cast(input: torch.Tensor) -> Tuple[torch.Tensor, torch.Tensor, torch.Tensor]:
+    """Helper function to make torch.svd work with other than fp32/64.
+
+    The function torch.svd is only implemented for fp32/64 which makes
+    impossible to be used by fp16 or others. What this function does, is cast
+    input data type to fp32, apply torch.svd, and cast back to the input dtype.
+
+    NOTE: in torch 1.8.1 this function is recommended to use as torch.linalg.svd
```

Comment:
In that case, we shall add a version check. Like if torch.__version__ > 1 8.0. Otherwise, there gonna be some user warnings.

Reply:
will add warning